astropy/uncertainty/tests/test_distribution.py

```diff
@@ -0,0 +1,305 @@
+        assert_quantity_allclose(self.distr.pdf_mad, expected)
+        assert_quantity_allclose(self.distr.pdf_smad, self.distr.pdf_mad * SMAD_FACTOR, rtol=1e-5)
+        assert_quantity_allclose(self.distr.pdf_smad, [3, 2, 4, 5] * self.distr.unit, rtol=0.05)
+
+        # make sure the right type comes out - should be a Quantity because it's
+        # now a summary statistic
+        assert not isinstance(self.distr.pdf_mad, Distribution)
+        assert isinstance(self.distr.pdf_mad, u.Quantity)
+        assert not isinstance(self.distr.pdf_smad, Distribution)
+        assert isinstance(self.distr.pdf_smad, u.Quantity)
+
+    def test_percentile(self):
+        expected = np.percentile(self.data, [10, 50, 90], axis=-1) * self.distr.unit
+        percs = self.distr.pdf_percentiles([10, 50, 90])
+        assert_quantity_allclose(percs, expected)
+        assert percs.shape == (3, 4)
+
+        # make sure the right type comes out - should be a Quantity because it's
+        # now a summary statistic
+        assert not isinstance(percs, Distribution)
+        assert isinstance(percs, u.Quantity)
+
+    def test_add_quantity(self):
+        distrplus = self.distr + [2000, 0, 0, 500] * u.pc
+        expected = (np.median(self.data, axis=-1) + np.array([2, 0, 0, 0.5])) * self.distr.unit
+        assert_quantity_allclose(distrplus.pdf_median, expected)
+        expected = np.var(self.data, axis=-1) * self.distr.unit**2
+        assert_quantity_allclose(distrplus.pdf_var, expected)
+
+    def test_add_distribution(self):
+        another_data = (np.random.randn(4, 10000)
+                        * np.array([1000, .01, 80, 10])[:, np.newaxis]
+                        + np.array([2000, 0, 0, 500])[:, np.newaxis])
+        # another_data is in pc, but main distr is in kpc
+        another_distr = Distribution(another_data * u.pc)
+        combined_distr = self.distr + another_distr
+
+        expected = np.median(self.data + another_data/1000,
+                             axis=-1) * self.distr.unit
+        assert_quantity_allclose(combined_distr.pdf_median, expected)
+
+        expected = np.var(self.data + another_data/1000, axis=-1) * self.distr.unit**2
+        assert_quantity_allclose(combined_distr.pdf_var, expected)
+
+
+def test_helper_normal_samples():
+    centerq = [1, 5, 30, 400] * u.kpc
+
+    with NumpyRNGContext(12345):
+        n_dist = ds.normal(centerq, std=[0.2, 1.5, 4, 1]*u.kpc, n_samples=100)
+        assert n_dist.distribution.shape == (4, 100)
+        assert n_dist.shape == (4, )
+        assert n_dist.unit == u.kpc
+        assert np.all(n_dist.pdf_std > 100*u.pc)
+
+        n_dist2 = ds.normal(centerq, std=[0.2, 1.5, 4, 1]*u.pc, n_samples=20000)
+        assert n_dist2.distribution.shape == (4, 20000)
+        assert n_dist2.shape == (4, )
+        assert n_dist2.unit == u.kpc
+        assert np.all(n_dist2.pdf_std < 100*u.pc)
+
+
+def test_helper_poisson_samples():
+    centerqadu = [1, 5, 30, 400] * u.adu
```

Comment:
Sorry for being anal, but perhaps use `u.count`? (Which also suggests that both `count` and `electron` should be explicitly allowed as units for poisson - thought I'd quite happily forego units for now...)